src/main/java/org/team1540/robot2025/subsystems/arm/ArmIOSim.java

```diff
@@ -0,0 +1,73 @@
+package org.team1540.robot2025.subsystems.arm;
+
+import static org.team1540.robot2025.Constants.LOOP_PERIODIC_SECS;
+import static org.team1540.robot2025.subsystems.arm.ArmConstants.*;
+
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+
+public class ArmIOSim implements ArmIO {
+
+    // fields
+    private final SingleJointedArmSim armSim = new SingleJointedArmSim(
+            DCMotor.getKrakenX60(1),
+            GEAR_RATIO,
+            ARM_MOMENT_OF_INERTIA,
+            ARM_LENGTH_METERS,
+            MIN_ANGLE.getRadians(),
+            MAX_ANGLE.getRadians(),
+            true,
+            STARTING_ANGLE.getRadians());
+
+    private double armAppliedVolts = 0.0;
+    private final ProfiledPIDController controller =
+            new ProfiledPIDController(KP, KI, KD, new TrapezoidProfile.Constraints(MAX_VELOCITY, MAX_ACCELERATION));
+    private final ArmFeedforward feedforward = new ArmFeedforward(KS, KG, KV);
+    private boolean isClosedLoop;
+    private TrapezoidProfile.State goalState;
+
+    // methods
+
+    @Override
+    public void updateInputs(ArmIOInputs inputs) {
+        if (isClosedLoop) {
+            armAppliedVolts = controller.calculate(armSim.getAngleRads(), inputs.position.getRadians())
+                    + feedforward.calculate(
+                            Units.rotationsToDegrees(controller.getSetpoint().position),
+                            controller.getSetpoint().velocity);
```

Comment:
These aren't the right units. The `ProfiledPIDController` takes units of rotations since the PID constants and max velocity and acceleration are based on rotations. `ArmFeedforward` always uses units of radians.